frontend/test/metabase/scenarios/datasets/helpers/e2e-datasets-helpers.js

```diff
@@ -0,0 +1,128 @@
+import { popover, modal } from "__support__/e2e/cypress";
+
+export function assertQuestionIsBasedOnDataset({
+  questionName,
+  collection,
+  dataset,
+  table,
+}) {
+  if (questionName) {
+    cy.findByText(questionName);
+  }
+
+  // Asserts shows dataset and its collection names
+  // instead of db + table
+  cy.findAllByText(collection);
+  cy.findByText(dataset);
+
+  cy.findByText("Sample Dataset").should("not.exist");
+  cy.findByText(table).should("not.exist");
+}
+
+export function assertCreatedNestedQuery(datasetId) {
+  cy.wait("@createCard").then(({ request }) => {
+    expect(request.body.dataset_query.query["source-table"]).to.equal(
+      `card__${datasetId}`,
+    );
+  });
+}
+
+export function saveQuestionBasedOnDataset({ datasetId, name }) {
+  cy.intercept("POST", "/api/card").as("createCard");
+
+  cy.findByText("Save").click();
+
+  modal().within(() => {
+    cy.findByText(/Replace original question/i).should("not.exist");
+    if (name) {
+      cy.findByLabelText("Name")
+        .clear()
+        .type(name);
+    }
+    cy.findByText("Save").click();
+  });
+
+  assertCreatedNestedQuery(datasetId);
+
+  modal()
+    .findByText("Not now")
+    .click();
+}
+
+export function selectDimensionOptionFromSidebar(name) {
+  cy.get("[data-testid=dimension-list-item]")
+    .contains(name)
+    .click();
+}
+
+export function openDetailsSidebar() {
+  cy.findByTestId("saved-question-header-button").click();
+}
+
+export function getDetailsSidebarActions() {
+  return cy.findByTestId("question-action-buttons");
+}
```

Comment:
These functions can be used for questions too. Would it make sense to move them to something like `helpers/e2e-question-helpers`?